receiver/k8sclusterreceiver/internal/hierarchicalresourcequota/hierarchicalresourcequotas.go

```diff
@@ -0,0 +1,73 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package hierarchicalresourcequota // import "github.com/open-telemetry/opentelemetry-collector-contrib/receiver/k8sclusterreceiver/internal/hierarchicalresourcequota"
+
+import (
+	"strings"
+
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.uber.org/zap"
+	"k8s.io/apimachinery/pkg/api/resource"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/receiver/k8sclusterreceiver/internal/metadata"
+)
+
+func RecordMetrics(logger *zap.Logger, mb *metadata.MetricsBuilder, hrq *unstructured.Unstructured, ts pcommon.Timestamp) {
```

Comment:
I'll try to use the following function to implement it.
```
informer.SetTransform(transformObject)
```

Reply:
> No, let's not add logic here, it's only for saving memory by removing unneeded fields. See the comment above the function:

OK.
I will implement a transform logic in collector.go#L109 instead of transformObject fuction.

```go
	dc.metadataStore.ForEach(gvk.HierarchicalResourceQuota, func(o any) {
                 hrq := hrq.Transform(o.(*unstructured.Unstructured)) // transform to v1alpha2.HierarchicalResourceQuota
		hierarchicalresourcequota.RecordMetrics(dc.settings.Logger, dc.metricsBuilder, hrq, ts)
	})
```